### PR TITLE
add android small demo

### PR DESCRIPTION
**Pull request by BreakZero**
use wallet-core-2.6.23 add a small demo. 
`requestAccounts` and `signMessage` checked

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -33,13 +33,18 @@ android {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
     implementation 'androidx.core:core-ktx:1.2.0'
     implementation 'androidx.appcompat:appcompat:1.2.0'
     implementation 'com.google.android.material:material:1.1.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation project(path: ':lib')
+
+    implementation("com.louiscad.splitties:splitties-alertdialog-material:3.0.0")
+
+    // implementation project(path: ':lib')
```
Comment (hewigovens) [inline android/app/build.gradle:45]:
```suggestion
```

File: android/app/build.gradle [hunk 2]
```diff
@@ -33,13 +33,18 @@ android {
 }
 
 dependencies {
-
+    implementation fileTree(dir: 'libs', include: ['*.jar', '*.aar'])
```
Comment (hewigovens) [inline android/app/build.gradle:36]: could you please install wallet core from GitHub package? don't commit binary aar directly into this repo

File: android/app/src/main/java/com/trust/web3/demo/WebAppInterface.kt [hunk 3]
```diff
@@ -5,26 +5,87 @@ import android.webkit.JavascriptInterface
...
-                val callback = "window.ethereum.sendResponse($id, [\"$addr\"])"
-                context.post {
-                    context.evaluateJavascript(callback) { value ->
-                        println(value)
+        val id = obj.getLong("id")
+        val method = DAppMethod.fromValue(obj.getString("name"))
+        when (method) {
+            DAppMethod.REQUESTACCOUNTS -> {
+                context.materialAlertDialog {
+                    title = "require account"
```
Comment (hewigovens) [inline android/app/src/main/java/com/trust/web3/demo/WebAppInterface.kt:30]:
```suggestion
                    title = "Request Accounts"
```

File: android/app/src/main/java/com/trust/web3/demo/WebAppInterface.kt [hunk 4]
```diff
@@ -5,26 +5,87 @@ import android.webkit.JavascriptInterface
...
+                    message = "dapp get your address"
```
Comment (hewigovens) [inline android/app/src/main/java/com/trust/web3/demo/WebAppInterface.kt:31]: show dapp's url ?

File: android/app/src/main/java/com/trust/web3/demo/X.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package com.trust.web3.demo
```
Comment (hewigovens) [inline android/app/src/main/java/com/trust/web3/demo/X.kt:1]: could you please rename `X.kt` to `WebViewExtension.kt`

**Review by hewigovens — COMMENTED** [review]
Thanks for the contribution, overall looks good

**Comment by BreakZero** [comment]
hi @hewigovens updated

**Comment by BreakZero** [comment]
hi @hewigovens updated

**Review by hewigovens — APPROVED** [review]
Thanks for the contribution

File: android/app/build.gradle [hunk 6]
```diff
@@ -33,13 +33,16 @@ android {
...
     implementation 'androidx.core:core-ktx:1.2.0'
     implementation 'androidx.appcompat:appcompat:1.2.0'
     implementation 'com.google.android.material:material:1.1.0'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation project(path: ':lib')
+
+    implementation "com.louiscad.splitties:splitties-alertdialog-material:3.0.0"
+    implementation "com.trustwallet:wallet-core:2.6.23"
+
+    implementation 'com.github.TrustWallet:trust-web3-provider:1.0.4'
```
Comment (vikmeup) [inline android/app/build.gradle:45]:
```suggestion
    implementation 'com.github.trustwallet:trust-web3-provider:1.0.4'
```

Comment (vikmeup) [reply android/app/build.gradle:45]: would this work?

**Comment by hewigovens** [comment]
@BreakZero thanks again, I refactored a bit and added personal sign and typed message support here: https://github.com/trustwallet/trust-web3-provider/pull/177

**Pull request closed without merging**